### ISSUE TITLE
Required reason API warning on version 10.24

### ISSUE THREAD

**Issue by lpikorhcp** [firebase/firebase-ios-sdk#12813]
### Description

I'm using the latest version `10.24.0`.
I'm importing following libraries:

```
Firebase
FirebaseAnalytics
FirebaseCrashlytics
FirebasePerformance
```

I still got the warning from the App Store:

> ITMS-91053: Missing API declaration - Your app’s code in the “XYZ” file references one or more APIs that require reasons, including the following API categories: NSPrivacyAccessedAPICategoryDiskSpace. While no action is required at this time, starting May 1, 2024, when you upload a new app or app update, you must include a NSPrivacyAccessedAPITypes array in your app’s privacy manifest to provide approved reasons for these APIs used by your app’s code. For more details about this policy, including a list of required reason APIs and approved reasons for usage, visit: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api.

I double-checked, I'm not using any API from `NSPrivacyAccessedAPICategoryDiskSpace` section. However, what I noticed is that Firebase SDK has this comment:

https://github.com/firebase/firebase-ios-sdk/blob/d58541efcaee7816b4240a525a20f101a167bdf4/Crashlytics/Crashlytics/Components/FIRCLSHost.m#L186-L188

`statfs` is a required reason API ([link](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278397)).

I opened all `PrivacyInfo.xcprivacy` files that are part of the Firebase SDK. There are 12 of them currently. 
None of the files says that is uses `NSPrivacyAccessedAPICategoryDiskSpace`. 
`NSPrivacyAccessedAPICategoryUserDefaults, NSPrivacyAccessedAPICategoryFileTimestamp` are the ones being mentioned.

Is it possible that App Store Connect does simple text-based search and that's the reason why my app is getting flagged with the usage of `NSPrivacyAccessedAPICategoryDiskSpace`?

### Firebase SDK Version

10.24.0

### Xcode Version

15.2

### Installation Method

Swift Package Manager

### Firebase Product(s)

Analytics, Crashlytics, Performance

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by rizafran** [comment]
Thanks for reporting, @lpikorhcp. This issue seems related to #12557.

**Comment by ncooke3** [comment]
> Is it possible that App Store Connect does simple text-based search and that's the reason why my app is getting flagged with the usage of NSPrivacyAccessedAPICategoryDiskSpace?

@lpikorhcp, I don't think so because the comment wouldn't end up in the binary. As @rizafran said above, the warnings are related to https://github.com/firebase/firebase-ios-sdk/issues/12557, but I'd like to better understand how the `NSPrivacyAccessedAPICategoryDiskSpace` warning is occurring. Could you share your `Package.resolved`?

**Comment by devssun** [comment]
Firebase's dependency library - GTMSessionFetcher 3.3.1 version hasn't NSPrivacyAccessedAPICategoryFileTimestamp API.
but that library's latest version has NSPrivacyAccessedAPICategoryFileTimestamp.

So I think Firebase need to update(?) dependency library version. please check.

**Comment by paulb777** [comment]
@devssun Firebase's GTMSessionFetcher dependency allows minor version updates in both CocoaPods and SPM, so run `pod update` or Update to Latest Packages depending on your package manager

**Comment by devssun** [comment]
@paulb777 Thanks 😊

**Comment by lpikorhcp** [comment]
@ncooke3 I found what was causing this issue for me, it was `StripeTerminal`

https://github.com/stripe/stripe-terminal-ios/blob/23e2ed54d808f8f6016d2b23ee5a40351a6535dd/Example/StripeTerminal.xcframework/ios-arm64/StripeTerminal.framework/PrivacyInfo.xcprivacy#L44-L49

I think this issue can be closed, considering that much more info is available at #12557.
